ui/proposals_page.go

```diff
@@ -308,23 +234,56 @@ func (pg *proposalsPage) layoutTabs(gtx C) D {
 	})
 }
 
-func (pg *proposalsPage) layoutFetchingState(gtx C) D {
-	str := "Fetching " + strings.ToLower(proposalCategoryTitles[pg.tabs.selected]) + " proposals..."
-	gtx.Constraints.Min.X = gtx.Constraints.Max.X
-	return layout.Center.Layout(gtx, func(gtx C) D {
-		return pg.theme.Body1(str).Layout(gtx)
-	})
+func (pg *proposalsPage) addDiscoveredProposal(proposal dcrlibwallet.Proposal) {
+	for i := range pg.tabs.tabs {
+		if pg.tabs.tabs[i].category == proposal.Category {
+			item := proposalItem{
+				btn:               new(widget.Clickable),
+				proposal:          proposal,
+				voteBar:           pg.theme.VoteBar(pg.infoIcon, pg.legendIcon),
+				infoIcon:          pg.infoIcon,
+				stateInfoTooltip:  pg.theme.Tooltip(),
+				stateTooltipLabel: pg.theme.Caption(""),
+			}
+			pg.tabs.tabs[i].proposals = append([]proposalItem{item}, pg.tabs.tabs[i].proposals...)
+			break
+		}
+	}
 }
 
-func (pg *proposalsPage) layoutErrorState(gtx C) D {
-	str := "Error fetching proposals"
+func (pg *proposalsPage) updateProposal(proposal dcrlibwallet.Proposal) {
```

Comment:
Kindly add a comment here or a more descriptive method name. What about the proposal is being updated?